lib/plugins/aws/customResources/resources/apiGatewayCloudWatchRole/handler.js

```diff
@@ -19,8 +18,9 @@ async function create(event, context) {
   const { RoleArn } = event.ResourceProperties;
   const { Partition: partition, AccountId: accountId, Region: region } = getEnvironment(context);
 
-  const apiGateway = new ApiGateway({ region });
-  const assignedRoleArn = (await apiGateway.getAccount().promise()).cloudwatchRoleArn;
+  const assignedRoleArn = (
+    await awsRequest({ name: 'APIGateway', params: { region } }, 'getAccount').promise()
```

Comment:
`awsRequest` should return promise directly, there should be no need for `promise()`